app/main.py

```diff
@@ -2,0 +1,37 @@
+class Animal:
+    alive = []
+
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False
+                 ) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def death(self) -> None:
+        if self in Animal.alive:
+            Animal.alive.remove(self)
+
+    def __repr__(self) -> str:
+        return (
+            f"{{Name: {self.name}, "
+            f"Health: {self.health}, "
+            f"Hidden: {self.hidden}}}"
+        )
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
```

Comment:
Code Efficiency issue: You can change the `bool` value in one line. Instead of `if self.hidden: self.hidden = False else: self.hidden = True` use `self.hidden = not self.hidden`.